packages/apps/reputation-oracle/server/src/modules/cron-job/cron-job.service.spec.ts

```diff
@@ -1,830 +1,369 @@
-      const cronJobType = CronJobType.ProcessPendingIncomingWebhook;
-
-      const cronJobEntity = new CronJobEntity();
-      cronJobEntity.cronJobType = cronJobType;
-      cronJobEntity.startedAt = new Date();
-
-      jest.spyOn(repository, 'findOneByType').mockResolvedValue(null);
+    it('should create a new cron job if none exists', async () => {
+      cronJobRepository.findOneByType.mockResolvedValue(null);
+      cronJobRepository.createUnique.mockResolvedValue(new CronJobEntity());
 
-      const createUniqueSpy = jest
-        .spyOn(repository, 'createUnique')
-        .mockResolvedValue(cronJobEntity);
-
-      const result = await service.startCronJob(cronJobType);
+      const result = await service.startCronJob(
+        CronJobType.ProcessPendingIncomingWebhook,
+      );
 
-      expect(createUniqueSpy).toHaveBeenCalledWith({
-        cronJobType: CronJobType.ProcessPendingIncomingWebhook,
-      });
-      expect(result).toEqual(cronJobEntity);
+      expect(cronJobRepository.findOneByType).toHaveBeenCalledWith(
+        CronJobType.ProcessPendingIncomingWebhook,
+      );
+      expect(cronJobRepository.createUnique).toHaveBeenCalled();
+      expect(result).toBeInstanceOf(CronJobEntity);
     });
 
-    it('should start a cron job if exists', async () => {
-      const cronJobType = CronJobType.ProcessPendingIncomingWebhook;
-      const cronJobEntity: Partial<CronJobEntity> = {
-        cronJobType: cronJobType,
+    it('should update an existing cron job', async () => {
+      const existingCronJob = new CronJobEntity();
+      existingCronJob.startedAt = new Date();
+      cronJobRepository.findOneByType.mockResolvedValue(existingCronJob);
+
+      const updatedCronJob = {
+        ...existingCronJob,
         startedAt: new Date(),
-        completedAt: new Date(),
+        completedAt: null,
       };
+      cronJobRepository.updateOne.mockResolvedValue(updatedCronJob as any);
 
-      const mockDate = new Date(2023, 12, 23);
-      jest.useFakeTimers();
-      jest.setSystemTime(mockDate);
-
-      const findOneByTypeSpy = jest
-        .spyOn(repository, 'findOneByType')
-        .mockResolvedValue(cronJobEntity as any);
-
-      const updateOneSpy = jest
-        .spyOn(repository, 'updateOne')
-        .mockResolvedValue(cronJobEntity as any);
-
-      const result = await service.startCronJob(cronJobType);
-
-      expect(findOneByTypeSpy).toHaveBeenCalledWith(cronJobType);
-      expect(updateOneSpy).toHaveBeenCalled();
-      cronJobEntity.startedAt = mockDate;
-      expect(result).toEqual(cronJobEntity);
+      const result = await service.startCronJob(
+        CronJobType.ProcessPendingIncomingWebhook,
+      );
 
-      jest.useRealTimers();
+      expect(cronJobRepository.findOneByType).toHaveBeenCalledWith(
+        CronJobType.ProcessPendingIncomingWebhook,
+      );
+      expect(cronJobRepository.updateOne).toHaveBeenCalledWith(updatedCronJob);
```

Comment:
Under the hood `startCronJob` updates `startedAt = new Date()`, so this condition might be flaky if tests run slowly.

Let's either fake timers or use `expect.objectContaining` and not check the exact value, but up to some second